JobScheduler.js

```diff
@@ -1,84 +1,215 @@
 'use strict';
```

Comment:
1. No - Every feature in the system has some pre-requisites. For scheduled updates the pre-req is that DB must be configured & serviceinstanceautoupdate job must be enabled. [See here](https://github.com/jagadish-kb/service-fabrik-broker/blob/master/lib/utils/index.js#L266). 
Now backup / restore are cloud qualities that are to be implemented & need not be tied to each and every feature that needs DB. Now if backup/restore does not exist/work, then it is a risk per-se, but its not a true dependency. Also in case of schedule update, we dont store any critical information. Just random schedules / run history. So even if DB crashes and we cant recover, its not too critical.

2. Polling is a feature for background jobs. If some one tries operations from UI / Cmd line, then they are provided with response to their request immediately. Don't need any polling for UI / Cmd line operations. The reason why background jobs are stopped from running during maintenance is because, if a particular job run fails then it will not run till its next scheduled time. However if an operation fails when user initiates he can always trigger it after sometime.  Even if system is in maintenance & if user initiates an operation then either broker times out or broker returns back successfully. If its asynch operations bosh takes care of performing that operation and cf keeps pining with last operation and if its docker related its going to be synch response.
So really don't see any reasons to block user from performing operations.

Reply:
Ok,thanks for the detailed explanation.